pkg/interfaces/contracts/liquidity-mining/IStakelessGaugeController.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IStakelessGauge.sol";
+
+/**
+ * @title Stakeless Gauge Controller interface
+ * @notice Manages checkpoints for registered stakeless gauges, allowing to perform mutiple checkpoints in a
+ * single call.
+ * @dev Supports Polygon, Arbitrum, Optimism, Gnosis and ZKSync stakeless root gauges. Gauges to be checkpointed need
+ * to be registered (added) to the controller beforehand.
+ */
+interface IStakelessGaugeController {
+
+    /**
+    * @dev Emitted when stakeless gauges are added to the controller.
+    */
+    event GaugesAdded(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] gauges);
+
+    /**
+    * @dev Emitted when stakeless gauges are removed from the controller.
+    */
+    event GaugesRemoved(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] gauges);
```

Comment:
It's better to a emit one event per gauge as opposed to including the full array. We should also make both the gauge type and the gauge itself `indexed`.

Reply:
Ack; done!